### ISSUE TITLE
Save more detailed data to enable data analysis

### ISSUE THREAD

**Issue by royitaqi** [royitaqi/HollowKnight.GodhomeWinLossTracker#10]
* [x] Figure out how to save to a separate folder (because data can be large)
* [x] Save each win/loss entries
  * [x] Date/time (local)
  * [x] Sequence
  * [x] Boss name
  * [x] Scene name
  * [x] Win and loss counts (not boolean) b3fe830dd6af4e4dbb64c69e274898c33e04d1d4
  * [x] Fight length (time between scenes)
* [x] Figure out what's up with save0
  * [x] Implemented hack to save profile ID into local data and use it to save/load folder data. 0d2b556f1223660927d3f51936123720cd7284ed\
  * [x] Try `PlayerData.instance.profileID`.
    * [x] Tried in 81b8224c1cc78f2a57647074bd5976fa608449cb. It's almost same as `GameManager.instance.profileID`.
    * [x] Tried again in bdb6fcf74df67259fc8fed70894c967ed74aaa54. **`PlayerData.instance.profileID` lags to show the last loaded profile ID, which can be misleading when loading a save.** E.g. load 3, save, then load 1. At the time of loading 1, `PlayerData.instance.profileID=3` and `GameManager.instance.profileID=0`.
* [x] Implement a "Export TSV" menu to write out TSV into the same save folder
  * [x] Change SaveLocalData message: add a format enum which can be JSON and TSV. Normal saves will go with JSON. Export will go with TSV. 
  * [x] Implement the export logic. Filename can be "{mod}/Export.Save1.txt".
  * [x] P1 display a notification containing the filename on successful export.

**Comment by royitaqi** [comment]
How to save to a separate folder:
See `QuickSettings.cs`

```
private static readonly string _baseDataPath = Application.persistentDataPath + '/';

private string SettingsDirectory => _baseDataPath + ModName + "." + nameof(QuickSettings) + '/';


        public string? GetSaveSettingsPath(int? saveSlotID = null)
        {
            if (saveSlotID == null && CurrentSaveSlot == null)
                return null;

            return SettingsDirectory + string.Format(SettingsSaveFileName, saveSlotID ?? CurrentSaveSlot);
        }


        public void SaveSaveSettings()
        {
            EnsureSettingsDirectory();
            string filePath = GetSaveSettingsPath()!;
            try
            {
                SerializeAndSave(filePath, _saveSettings);
            }
            ...
        }


        protected void Initialize(string? modName)
        {
            if (modName == null)
                throw new ArgumentNullException(nameof(modName), $"Couldn't find the name of the mod to use. Try using '{nameof(QuickSettings)}(Type)'.");

            ModName = modName;

            _owningAssembly ??= Assembly.GetCallingAssembly();
            RegisterDefinedSettings();
            LoadGlobalSettings();
            AddHooks();

            if (GameManager.instance && GameManager.instance.gameState is GameState.PLAYING or GameState.PAUSED)
            {
                CurrentSaveSlot = GameManager.instance.profileID;
                LoadSaveSettings();
            }

            Initialized?.Invoke();
        }


        public void Unload()
        {
            RemoveHooks();
            SaveAllSettings();
            Unloaded?.Invoke();
        }
```

How to use:
```
// in BindingManager.cs
        // per-save
        [QuickSetting(true, nameof(RegisteredBindings))]
        private static List<Binding> _serializedBindings = new();

        [QuickSetting]
        public static ActionCombo OpenBindingsUI { get; private set; }

        [QuickSetting, DefaultValue(true)]
        internal static bool EnforceBindingRestrictions { get; private set; } = true;

        [QuickSetting]
        public static bool AllowEssentialCharms { get; private set; } = true;

        [QuickSetting("EssentialCharms")]
        internal static int[] ExemptCharms =
        {
            36, // Void Heart
            40, // Grimmchild
        };

        [QuickSetting]
        public static int MaxBoundNailDamage { get; private set; } = 13;

        [QuickSetting]
        public static int MaxBoundShellHealth { get; private set; } = 4;


// in ToggleableBindings: Mod
        public override void Initialize(Dictionary<string, Dictionary<string, GameObject>> preloadedObjects)
        {
            AddHooks();
            Settings = new(); // <--- This is important.
            BindingManager.Initialize();
            BindingsUIController.Initialize();

            LogDebug("Initialized.");
        }

```

**Comment by royitaqi** [comment]
Implemented in 3dedd7d56b73df15ecc0610d41ad5531bdfd7e03
